pytorch_lightning/accelerators/tpu.py

```diff
@@ -49,3 +51,11 @@ def run_optimizer_step(
         self, optimizer: Optimizer, optimizer_idx: int, lambda_closure: Callable, **kwargs: Any
     ) -> None:
         xm.optimizer_step(optimizer, optimizer_args={'closure': lambda_closure, **kwargs})
+
+    def _move_optimizer_state(self, device: Optional[torch.device] = None) -> None:
+        """ Moves the state of the optimizers to the TPU if needed. """
+        # TODO: `self.root_device` would raise error if called outside the spawn process
+        # while training on 8 and more cores.
+        for opt in self.optimizers:
+            for p, v in opt.state.items():
+                opt.state[p] = apply_to_collection(v, torch.Tensor, move_data_to_device, self.root_device)
```

Comment:
@kaushikb11 here you are calling `self.root_device` anyway, despite the comment above?

Reply:
Yes, not sure to understand the reasoning too.